src/libcore/macros.rs

```diff
@@ -108,6 +108,44 @@ macro_rules! assert_eq {
     });
 }
 
+/// Asserts that two expressions are not equal to each other.
+///
+/// On panic, this macro will print the values of the expressions with their
+/// debug representations.
+///
+/// # Examples
+///
+/// ```
+/// let a = 3;
+/// let b = 2;
+/// assert_ne!(a, b);
+/// ```
+#[macro_export]
+///#[stable(feature = "????", since = "????")]
+macro_rules! assert_eq {
+    ($left:expr , $right:expr) => ({
+        match (&$left, &$right) {
+            (left_val, right_val) => {
+                if *left_val == *right_val {
+                    panic!("assertion failed: `(left != right)` \
```

Comment:
i think this is correct based on the pattern established by `assert_eq` (https://github.com/rust-lang/rust/blob/master/src/libcore/macros.rs#L103). though i'll agree it is a bit ambiguous sans the context of `assert_eq`'s impl.